server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/TextCalendarBodyToAttachmentTest.java

```diff
@@ -197,41 +150,57 @@ public void serviceShouldConvertTextBodyOfMessageToAttachmentWhenTextCalendar()
     }
 
     @Test
-    public void serviceShouldKeepAllMessageHeaderWhenConverting() throws Exception {
-        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(MESSAGE_CONTENT.getBytes()));
-        List<Header> allHeaders = Collections.list(message.getAllHeaders());
-        List<String> allStringHeaders = allHeaders.stream()
-            .map(Header::getName)
-            .collect(Guavate.toImmutableList());
+    public void serviceShouldKeepHeaderValueExceptContentTypeAndMessageIdAndEncodingWhenConverting() throws Exception {
+        List<String> headerChanged = ImmutableList.of("Content-Type", "Message-ID", "Content-transfer-encoding");
+        List<Header> allHeaders = Collections.list(calendarMessage.getAllHeaders());
 
         Mail mail = FakeMail.builder()
-            .mimeMessage(message)
+            .mimeMessage(calendarMessage)
             .build();
 
         mailet.service(mail);
 
         List<Header> actualHeaders = Collections.list(mail.getMessage().getAllHeaders());
-        List<String> actualStringHeaders = actualHeaders
-            .stream()
-            .map(Header::getName)
-            .collect(Guavate.toImmutableList());
+        for (Header header : allHeaders) {
+            if (!headerChanged.contains(header.getName())) {
+                assertThat(actualHeaders)
+                    .extracting(Header::getName, Header::getValue)
+                    .contains(Tuple.tuple(header.getName(), header.getValue()));
+            }
+        }
+    }
+
+    @Test
+    public void contentTypeOfAttachmentShouldBeTakenFromOriginalMessage() throws Exception {
+        String originalContentType = calendarMessage.getContentType();
 
-        assertThat(actualStringHeaders).containsAll(allStringHeaders);
+        Mail mail = FakeMail.builder()
+            .mimeMessage(calendarMessage)
+            .build();
+
+        mailet.service(mail);
+
+        Multipart multipart = (Multipart)mail.getMessage().getContent();
+
+        int firstBodyPartIndex = 0;
+        BodyPart firstBodyPart = multipart.getBodyPart(firstBodyPartIndex);
+        assertThat(firstBodyPart.getContentType()).isEqualTo(originalContentType);
     }
 
     @Test
-    public void contentTypeOfAttachmentShouldGetFromOriginalMessage() throws Exception {
-        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(MESSAGE_CONTENT.getBytes()));
+    public void contentTransferEncodingOfAttachmentShouldBeTakenFromOriginalMessage() throws Exception {
+        String originalContentTransferEncoding = calendarMessage.getEncoding();
```

Comment:
What is the value of originalContentTransferEncoding?

String originalContentTransferEncoding = "8-BIT";